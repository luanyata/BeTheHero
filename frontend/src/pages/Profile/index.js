import React from "react";

import { Link } from "react-router-dom";
import logoImg from "../../assets/logo.svg";

import "./style.css";
import { FiPower, FiTrash2 } from "react-icons/fi";

function Profile() {
  return (
    <div className="profile-container">
      <header>
        <img src={logoImg} alt="Be the Hero" />
        <span>Bem Vinda, APAD</span>

        <Link className="button" to="/incidents/new">
          Cadastrar novo caso
        </Link>
        <button>
          <FiPower sizr={18} color="#E02041" />
        </button>
      </header>
      <h1>Casos Cadastrados</h1>
      <ul>
        <li>
          <strong>CASO:</strong>
          <p>Caso teste</p>

          <strong>Descricao:</strong>
          <p>Descricao teste</p>

          <strong>Valor:</strong>
          <p>R$ 120,00</p>
          <button type="button">
            <FiTrash2 size={20} color="#A8A8B3" />
          </button>
        </li>

        <li>
          <strong>CASO:</strong>
          <p>Caso teste</p>

          <strong>Descricao:</strong>
          <p>Descricao teste</p>

          <strong>Valor:</strong>
          <p>R$ 120,00</p>
          <button type="button">
            <FiTrash2 size={20} color="#A8A8B3" />
          </button>
        </li>

        <li>
          <strong>CASO:</strong>
          <p>Caso teste</p>

          <strong>Descricao:</strong>
          <p>Descricao teste</p>

          <strong>Valor:</strong>
          <p>R$ 120,00</p>
          <button type="button">
            <FiTrash2 size={20} color="#A8A8B3" />
          </button>
        </li>

        <li>
          <strong>CASO:</strong>
          <p>Caso teste</p>

          <strong>Descricao:</strong>
          <p>Descricao teste</p>

          <strong>Valor:</strong>
          <p>R$ 120,00</p>
          <button type="button">
            <FiTrash2 size={20} color="#A8A8B3" />
          </button>
        </li>
      </ul>
    </div>
  );
}

export default Profile;
