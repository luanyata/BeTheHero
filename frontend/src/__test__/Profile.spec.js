import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import Profile from "../pages/Profile";

describe("<Profile />", () => {
  it("should to render the <Profile /> component", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
