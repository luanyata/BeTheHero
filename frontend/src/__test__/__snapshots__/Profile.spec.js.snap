// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should to render the <Profile /> component 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Profile>
      <div className=\\"profile-container\\">
        <header>
          <img src=\\"logo.svg\\" alt=\\"Be the Hero\\" />
          <span>
            Bem Vinda, 
          </span>
          <Link className=\\"button\\" to=\\"/incidents/new\\">
            <LinkAnchor className=\\"button\\" href=\\"/incidents/new\\" navigate={[Function: navigate]}>
              <a className=\\"button\\" href=\\"/incidents/new\\" onClick={[Function: onClick]}>
                Cadastrar novo caso
              </a>
            </LinkAnchor>
          </Link>
          <button>
            <FiPower sizr={18} color=\\"#E02041\\" onClick={[Function: handleLogout]}>
              <IconBase attr={{...}} sizr={18} color=\\"#E02041\\" onClick={[Function: handleLogout]}>
                <svg stroke=\\"currentColor\\" fill=\\"none\\" strokeWidth=\\"2\\" viewBox=\\"0 0 24 24\\" strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" sizr={18} color=\\"#E02041\\" onClick={[Function: handleLogout]} className={[undefined]} style={{...}} height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                  <path d=\\"M18.36 6.64a9 9 0 1 1-12.73 0\\" />
                  <line x1=\\"12\\" y1=\\"2\\" x2=\\"12\\" y2=\\"12\\" />
                </svg>
              </IconBase>
            </FiPower>
          </button>
        </header>
        <h1>
          Casos Cadastrados
        </h1>
        <ul />
      </div>
    </Profile>
  </Router>
</MemoryRouter>"
`;
